hs_core/tests/api/native/test_copy_resource.py

```diff
@@ -0,0 +1,294 @@
+
+        # test to make sure the new copied resource is linked with the original resource via
+        # isDerivedFrom Source metadata element
+        self.assertGreater(new_res_generic.metadata.sources.all().count(), 0,
+                           msg="New copied resource does not has source element.")
+
+        derived_from_value = '{}/resource/{}'.format(hydroshare.utils.current_site_url(),
+                                                     self.res_generic.short_id)
+        self.assertIn(derived_from_value,
+                      [src.derived_from for src in new_res_generic.metadata.sources.all()],
+                      msg="The original resource identifier is not set in isDerivedFrom Source "
+                          "metadata element of the new copied resource")
+
+    def test_copy_raster_resource(self):
+        # ensure a nonowner who does not have permission to view a resource cannot copy it
+        with self.assertRaises(PermissionDenied):
+            hydroshare.create_empty_resource(self.res_raster.short_id,
+                                             self.nonowner,
+                                             action='copy')
+        # give nonowner view privilege so nonowner can create a new copy of this resource
+        self.owner.uaccess.share_resource_with_user(self.res_raster, self.nonowner,
+                                                    PrivilegeCodes.VIEW)
+
+        new_res_raster = hydroshare.create_empty_resource(self.res_raster.short_id,
+                                                          self.nonowner,
+                                                          action='copy')
+        new_res_raster = hydroshare.copy_resource(self.res_raster, new_res_raster)
+
+        # test the new copied resource has the same resource type as the original resource
+        self.assertTrue(isinstance(new_res_raster, RasterResource))
+
+        # test science metadata elements are copied from the original resource to the new copied
+        # resource
+        self.assertTrue(new_res_raster.metadata.title.value == self.res_raster.metadata.title.value)
+        self.assertTrue(new_res_raster.creator == self.nonowner)
+
+        # test extended metadata elements are copied from the original resource to the new
+        # copied resource
+        if OriginalCoverage.objects.filter(object_id=self.res_raster.metadata.id).exists():
+            self.assertTrue(OriginalCoverage.objects.filter(
+                object_id=new_res_raster.metadata.id).exists())
+            self.assertEqual(new_res_raster.metadata.originalCoverage.value,
+                             self.res_raster.metadata.originalCoverage.value,
+                             msg="OriginalCoverage of new copied resource is not equal to "
+                                 "that of the original resource")
+
+        if CellInformation.objects.filter(object_id=self.res_raster.metadata.id).exists():
```

Comment:
@pkdash OK. Since CellInformation is a required field for raster resource type, agreed this check is not needed. Will remove this check then.